### ISSUE TITLE
Some APIs that support constrained JSON schema output require JSON schema in instructions

### ISSUE THREAD

**Issue by ajastro** [Arjun-Vashistha/Test-Repo-2#304]
### Description

Currently, `reponse_format` is set to json_object for PromptedOutput if json object is supported.

It's better to set it to json schema if json schema is supported just like native output, before checking the json_object.

The problem with NativeOutput is that, for many endpoints, enabling structured output alone without guiding in the prompt causes bad and unstable response. After all the model doesn't know the schema, it's just logits mask. Openai is a notable exception though.

I can contribute to this if the proposal is OK for the maintainers.

---
*Originally posted by @xcpky in pydantic/pydantic-ai*
*Upstream issue: https://github.com/pydantic/pydantic-ai/issues/3490*
